src/vizier/services/query_broker/script_runner/cloud_source.go

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package scriptrunner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/nats-io/nats.go"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/metadata"
+
+	"px.dev/pixie/src/shared/cvmsgspb"
+	"px.dev/pixie/src/shared/scripts"
+	svcutils "px.dev/pixie/src/shared/services/utils"
+	"px.dev/pixie/src/vizier/services/metadata/metadatapb"
+)
+
+// CloudSource is a Source that pulls cron scripts form the cloud
+type CloudSource struct {
+	scripts    map[string]*cvmsgspb.CronScript
+	stop       func()
+	client     metadatapb.CronScriptStoreServiceClient
+	nc         *nats.Conn
+	signingKey string
+}
+
+// NewCloudSource constructs a Source that will pull cron scripts from the cloud
+func NewCloudSource(nc *nats.Conn, csClient metadatapb.CronScriptStoreServiceClient, signingKey string) *CloudSource {
+	return &CloudSource{
+		client:     csClient,
+		nc:         nc,
+		signingKey: signingKey,
+	}
+}
+
+// Start subscribes to updates from the cloud on the CronScriptUpdatesChannel and sends resulting updates on updatesCh
+func (source *CloudSource) Start(baseCtx context.Context, updatesCh chan<- *cvmsgspb.CronScriptUpdate) error {
+	ctx := metadata.AppendToOutgoingContext(baseCtx,
+		"authorization", fmt.Sprintf("bearer %s", cronScriptStoreToken(source.signingKey)),
+	)
+	sub, err := source.nc.Subscribe(CronScriptUpdatesChannel, natsUpdater(ctx, source.nc, source.client, updatesCh))
+	if err != nil {
+		return err
+	}
+	unsubscribe := func() {
+		if err := sub.Unsubscribe(); err != nil {
+			log.WithError(err).Error("could not unsubscribe from cloud cron script updates")
+		}
+	}
+	initialScripts, err := fetchInitialScripts(ctx, source.nc, source.client)
+	if err != nil {
+		unsubscribe()
+		return err
+	}
+
+	source.scripts = initialScripts
+	source.stop = unsubscribe
+	return nil
+}
+
+// GetInitialScripts returns the initial set of scripts that all updates will be based on
+func (source *CloudSource) GetInitialScripts() map[string]*cvmsgspb.CronScript {
+	return source.scripts
```

Comment:
Rather than storing `source.scripts`, since we don't keep `source.scripts` up-to-date as new scripts roll in, why don't we call `fetchInitialScripts` in here?

Alternatively, if we always call `GetInitialScripts` right after `Start`, maybe start should just return the initialScripts.